omegaconf/_impl.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any
+
+from omegaconf import MISSING, Container, DictConfig, ListConfig, Node, ValueNode
+from omegaconf.errors import InterpolationToMissingValueError
+
+
+def _resolve_container_value(cfg: Container, node: Node, index: Any) -> None:
+    if node._is_interpolation():
+        try:
+            resolved = node._dereference_node()
+            assert resolved is not None
+            if isinstance(resolved, Container) and isinstance(node, ValueNode):
+                cfg[index] = resolved
+            else:
+                node._set_value(resolved._value())
+        except InterpolationToMissingValueError:
+            node._set_value(MISSING)
```

Comment:
This code isn't recursively resolving deeper interpolations. Here's a failure case:
```python
cfg = OmegaConf.create({"x": "${y}", "y": {"z": "${foo}"}, "foo": 0})
OmegaConf.resolve(cfg)
cfg  # <== {'x': {'z': '${foo}'}, 'y': {'z': 0}, 'foo': 0}
```